packages/insomnia/src/network/o-auth-2/get-token.ts

```diff
@@ -159,7 +159,7 @@ export const getOAuth2Token = async (
       ...insertAuthKeyIf('client_secret', authentication.clientSecret),
     ];
   } else {
-    headers.push(getBasicAuthHeader(authentication.clientId, authentication.clientSecret));
+    headers.push(getBasicAuthHeader(encodeURIComponent(authentication.clientId ?? ''), encodeURIComponent(authentication.clientSecret ?? '')));
```

Comment:
It would make more sense to put the _escaping_ in the `getBasicAuthHeader` function, so every usage will have it.
(currently there are 9 usage of the function)

Reply:
Yes, you're right. My bad !

> The client identifier is encoded using the "application/x-www-form-urlencoded" encoding algorithm per [Appendix B](https://datatracker.ietf.org/doc/html/rfc6749#appendix-B), and the encoded value is used as the username; the client password is encoded using the same algorithm and used as the password.

(https://datatracker.ietf.org/doc/html/rfc6749#section-2.3.1)